From cc1d2f0148e9c9bc4f1c2eb771cb65f1d025ebbe Mon Sep 17 00:00:00 2001
From: Jussi Kivilinna <jussi.kivilinna@iki.fi>
Date: Sat, 22 Jun 2024 15:51:06 +0300
Subject: [PATCH] src/hwf-x86: disable inline assembly CFI directivies for
 WIN32

* src/hwf-x86.c (CFI_ADJUST_CFA_OFFSET, CFI_PUSH4)
(CFI_POP4): Define only if _WIN32 is not defined.
--

Signed-off-by: Jussi Kivilinna <jussi.kivilinna@iki.fi>
---
 src/hwf-x86.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/hwf-x86.c b/src/hwf-x86.c
index bda14d9d..d9d34c0e 100644
--- a/src/hwf-x86.c
+++ b/src/hwf-x86.c
@@ -39,7 +39,7 @@
 #if defined (__i386__) && SIZEOF_UNSIGNED_LONG == 4 && defined (__GNUC__)
 # define HAS_X86_CPUID 1
 
-#ifdef HAVE_GCC_ASM_CFI_DIRECTIVES
+#if defined(HAVE_GCC_ASM_CFI_DIRECTIVES) && !defined(_WIN32)
 # define CFI_ADJUST_CFA_OFFSET(off) ".cfi_adjust_cfa_offset " #off "\n\t"
 # define CFI_PUSH4 CFI_ADJUST_CFA_OFFSET(4)
 # define CFI_POP4 CFI_ADJUST_CFA_OFFSET(-4)
-- 
2.43.0

